Mathlib/Analysis/Calculus/MeanValue.lean

```diff
@@ -596,6 +596,52 @@ theorem eqOn_of_fderivWithin_eq (hs : Convex ℝ s) (hf : DifferentiableOn 𝕜
   refine hs.is_const_of_fderivWithin_eq_zero (hf.sub hg) (fun z hz => ?_) hx hy
   rw [fderivWithin_sub (hs' _ hz) (hf _ hz) (hg _ hz), sub_eq_zero, hf' _ hz]
 
+/-- If `f` has zero derivative on an open set, then `f` is locally constant on `s`. -/
+theorem _root_.IsOpen.isOpen_inter_preimage_of_fderiv_eq_zero
+    (hs : IsOpen s) (hf : DifferentiableOn 𝕜 f s)
+    (hf' : ∀ x ∈ s, fderiv 𝕜 f x = 0) (t : Set G) : IsOpen (s ∩ f ⁻¹' t) := by
```

Comment:
Probably this won't affect much, but it's nice to use the dedicated spelling when it exists. Same applies elsewhere.
```suggestion
    (hf' : s.EqOn (fderiv 𝕜 f x) 0) (t : Set G) : IsOpen (s ∩ f ⁻¹' t) := by
```